src/inmanta/data/__init__.py

```diff
@@ -4930,166 +4930,186 @@ async def insert_sets_and_resources(
         environment: uuid.UUID,
         target_version: int,
         updated_resources: list[m.Resource],
-        updated_resource_sets: list[str | None],
+        updated_resource_sets: set[str | None],
```

Comment:
I could calculate this from the list of `updated_resources` . Do you think it makes sense to iterate over the resource list to fetch these resource_sets or can I just leave it as an argument?